here-naksha-lib-psql/src/commonMain/kotlin/naksha/psql/executors/write/DropCollection.kt

```diff
@@ -0,0 +1,51 @@
+package naksha.psql.executors.write
+
+import naksha.base.Platform.PlatformCompanion.logger
+import naksha.base.PlatformLogger
+import naksha.model.*
+import naksha.model.Naksha.NakshaCompanion.VIRT_COLLECTIONS
+import naksha.model.Naksha.NakshaCompanion.VIRT_COLLECTIONS_QUOTED
+import naksha.model.objects.NakshaCollection
+import naksha.psql.*
+import naksha.psql.executors.WriteExt
+
+class DropCollection(private val session: PgSession) {
+
+    fun execute(map: PgMap, write: WriteExt): TupleNumber {
+        if(write.collectionId != VIRT_COLLECTIONS){
+            throw NakshaException(
+                NakshaError.ILLEGAL_ARGUMENT,
+                "Expected $VIRT_COLLECTIONS collectionId when dropping collections"
+            )
+        }
+        val collectionId = write.featureId ?: throw NakshaException(
+            NakshaError.ILLEGAL_ARGUMENT,
+            "DROP without collectionId (expected in write's 'featureId')"
+        )
+        val pgCollection = PgCollection(map, collectionId)
+        val tupleNumber = collectionTupleNumber(pgCollection)
+        val conn = session.usePgConnection()
+        try {
+            pgCollection.drop(conn)
+            removeCollectionFromVirtualCollections(collectionId, conn)
+            conn.commit()
+            return tupleNumber
+        } catch (e: Exception) {
+            logger.info("Exception when dropping collection $collectionId, rolling back and throwing exception down the chain", e)
+            conn.rollback()
+            throw e
+        }
+    }
+
+    private fun removeCollectionFromVirtualCollections(collectionId: String, connection: PgConnection){
+        connection.execute(
```

Comment:
I'm fine with this, but please add a TODO, because this does not write history, and we do not keep track of this in the transaction log, but for now, I'm okay with this, we can fix later.

Reply:
added